src/DemoFile/DemoParser.cs

```diff
@@ -9,18 +9,25 @@ namespace DemoFile;
 
 public sealed partial class DemoParser
 {
+    private readonly ArrayPool<byte> _bytePool = ArrayPool<byte>.Create();
```

Comment:
please use shared ArrayPool, or allow the user to specify it (in constructor, or as a property)

Reply:
the reason is, shared ArrayPool (or custom provided) may already have ready buffers, so that way you avoid allocating new buffers

Reply:
The reason to not use the shared ArrayPool is that it is thread-safe, and the locking is unnecessary overhead. I can't think of other libraries that expose the ArrayPool that is used.

> the reason is, shared ArrayPool (or custom provided) may already have ready buffers, so that way you avoid allocating new buffers

This is true, but as long as the same few arrays are used during parsing, it doesn't really matter. We're talking allocations of a handful of arrays here over the whole demo

Reply:
afaik every ArrayPool is thread-safe : https://learn.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1?view=net-8.0#thread-safety

 but, you're right that there will be only a few arrays alive at any given time, so I guess it's fine

Reply:
speaking of which, I've seen some places where 16 length array is rented, this creates overhead, maybe it's better to use `Span<byte> span = stackalloc [16];` ?

Reply:
Can't stackalloc in async methods